### PR TITLE
Add bazel-diff to BCR

### PR DESCRIPTION
**Pull request by maxwellE**
Adds bazel-diff to the bcr

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (bazel-diff) have been updated in this PR. Please review the changes.

File: modules/bazel-diff/7.2.3/source.json
Comment (fmeum) [inline modules/bazel-diff/7.2.3/source.json:2]: Could you create a stable release archive?

Comment (maxwellE) [reply modules/bazel-diff/7.2.3/source.json]: Done

File: modules/bazel-diff/7.2.5/MODULE.bazel
Comment (fmeum) [inline modules/bazel-diff/7.2.5/MODULE.bazel:14]: Since bazel-diff is a tool, it shouldn't share the default Maven dependency closure. Could you set a different name, e.g. `bazel_diff_maven`?

**Comment by meteorcloudy** [comment]
@maxwellE Did you encounter https://github.com/Tinder/bazel-diff/issues/236? We recently want to remove completely WORKSPACE usage in Bazel but got blocked by this.

**Comment by maxwellE** [comment]
> @maxwellE Did you encounter [Tinder/bazel-diff#236](https://github.com/Tinder/bazel-diff/issues/236)? We recently want to remove completely WORKSPACE usage in Bazel but got blocked by this.

I will look at this next

File: modules/bazel-diff/7.2.13/MODULE.bazel
Comment (Wyverald) [inline modules/bazel-diff/7.2.13/MODULE.bazel:14]: this is not what @fmeum meant with a different name. What you call the variable is irrelevant -- you can call it `foo` and nothing would change. What you need to do is to explicitly pass `name = "bazel_diff_maven"` to the `install` tag, like this: https://github.com/bazelbuild/bazel-central-registry/blob/aadc90c53943809be4b6b93793acf82ab5877094/modules/copybara/0.0.0-20240326-6485af6/MODULE.bazel#L23

File: modules/bazel-diff/7.2.13/MODULE.bazel
Comment (Wyverald) [inline modules/bazel-diff/7.2.13/MODULE.bazel:31]: BTW, you can write `use_repo(maven, maven = "bazel_diff_maven")` here to avoid having to change all your references to `@maven` in your BUILD files. But it looks like you already changed those, so this is up to you.

File: modules/bazel-diff/7.2.14/source.json
Comment (fmeum) [inline modules/bazel-diff/7.2.14/source.json:2]: Is this file intentionally not compressed?

**Review by fmeum — DISMISSED** [review]

File: modules/bazel-diff/7.2.14/source.json
Comment (maxwellE) [reply modules/bazel-diff/7.2.14/source.json]: Fixed

**Review by fmeum — APPROVED** [review]

**Pull request merged**
